src/inmanta/deploy/state.py

```diff
@@ -87,6 +118,18 @@ class DeploymentResult(StrEnum):
     FAILED = enum.auto()
     SKIPPED = enum.auto()
 
+    @classmethod
+    def from_handler_resource_state(cls, handler_resource_state: const.HandlerResourceState) -> "DeploymentResult":
+        match handler_resource_state:
+            case const.HandlerResourceState.deployed:
+                return DeploymentResult.DEPLOYED
+            case const.HandlerResourceState.skipped | const.HandlerResourceState.skipped_for_dependency:
+                return DeploymentResult.SKIPPED
+            case const.HandlerResourceState.dry:
+                raise Exception(f"Unexpected handler_resource_state {const.HandlerResourceState.dry.value}")
```

Comment:
I wonder if we should return a fail an log a warning?

i.e. will this bork the scheduler?

Reply:
The deploy task will catch this exception and log it, it will also mark the deployment result as FAILED. This is not going to break the compiler.